dvc/project.py

```diff
@@ -167,7 +167,8 @@ def reproduce(self, targets, recursive=True, force=False):
         for target in targets:
             node = os.path.relpath(os.path.abspath(target), self.root_dir)
             if node not in stages:
-                raise StageNotFoundError(target)
+                self.logger.info(u'Stage file \'{}\' was not found'.format(node))
```

Comment:
We should actually collect stages first(from targets) and throw exception if something is not found. And only then proceed to reproducing all the collected stages.